mininet/node.py

```diff
@@ -866,10 +868,19 @@ def setHID( self, *hid ):
             self.cmd( 'xip hid new', p )
             self.cmd( 'xip hid addaddr', p )
 
+    def setZFID( self, *zfid ):
+        """Configure zFilter Identifier(s) for the host
+           zfid: A list of zFilter IDs"""
+        moduleDeps( add='xia_ppal_zf' )
+        for z in zfid:
+            if len( z ) == 40 and int( z, 2 ) > 0:
+                self.cmd( 'xip zf addlocal', z )
+
     def cleanup( self ):
         "Unload the modules loaded by the principals."
         moduleDeps( subtract='xia_ppal_hid' )
         moduleDeps( subtract='xia_ppal_xdp' )
+        moduleDeps( subtract='xia_ppal_zf' )
```

Comment:
Does this function cause an error message when trying to remove a module that is not loaded? If so, we may need to keep some state in `self` so that we can know which modules need to be unloaded.

Reply:
moduleDeps() only removes a module when it's loaded and ignores otherwise.

Reply:
OK, cool. It looks like the `subtract` parameter can be passed a list, so let's do that here instead:

    moduleDeps( subtract=['xia_ppal_hid', 'xia_ppal_xdp', 'xia_ppal_zf']